src/upsert.rs

```diff
@@ -34,12 +32,15 @@ pub trait FindexUpsert<
     /// - `master_key`  : Findex master key
     /// - `label`       : additional public information used in key hashing
     /// - `items`       : set of keywords used to index values
+    ///
+    /// Returns a map of keywords to booleans indicating whether the keyword
+    /// was already present in the database.
     async fn add(
         &mut self,
         master_key: &KeyingMaterial<MASTER_KEY_LENGTH>,
         label: &Label,
         items: HashMap<IndexedValue, HashSet<Keyword>>,
-    ) -> Result<(), Error<CustomError>> {
+    ) -> Result<HashMap<Keyword, bool>, Error<CustomError>> {
```

Comment:
You just need to return the set of keyword that were not part of the index before the `add` operation. This is lighter and you can get your boolean using `.contains()`.